charms/reactive/bus.py

```diff
@@ -419,7 +419,11 @@ def discover():
     for search_dir in ('reactive', 'hooks/reactive', 'hooks/relations'):
         search_path = os.path.join(hookenv.charm_dir(), search_dir)
         for dirpath, dirnames, filenames in os.walk(search_path):
+            if os.path.basename(dirpath) == '__pycache__':
+                continue
             for filename in filenames:
+                if filename.endswith('.pyc'):
```

Comment:
Do we need to worry about the other types of compiled Python files, like `*.pyo`?

Would it make sense to just allow files that either don't have an extensions, or have a `.sh` extension?

Reply:
Yeah, that seems reasonable.   TBH, I highly doubt anyone is actually using the ExternalHandler API at all, so making it more restrictive is unlikely to cause any issues.